### DISCUSSION TITLE
Using SynthMorph to register atlas label map to subject.

### DISCUSSION THREAD

**Discussion by andy9t7** [voxelmorph/voxelmorph#612] [Q&A] [answered]
Hi Voxelmorph/Synthmorph community,

I am trying to register an anatomical atlas (label map) of a specific subject to another subject. I was hoping to do this by first registering the atlas' underlying MRI to the target subjects MRI. I then apply its transform (via 'mri_convert') to the atlas label map to get it on the target subject.

1. Register atlas MRI to target MRI to get transform:
`mri_synthmorph -t transform.mgz atlas_MRI.mgz target_MRI.mgz`
2. Apply transform to atlas label map to get atlas 'registered' to target MRI:
`mri_convert -at transform.mgz atlas_labelmap.mgz transformed_atlas_labelmap.mgz`

What I found is that the MRI's register well using SynthMorph, however, when I apply the transform to the label map, the overall shape looks aligned but there is 'noisy' labels as shown in the images. Could I please get some guidance on how to fix this?
![Screenshot 2024-08-08 at 8 45 27 AM](https://github.com/user-attachments/assets/285306ef-bb2e-4fcf-b556-d2a67e6cdee2)

Andy

**Comment by adalca** [comment]
I'll ping @mu40 , although that looks like label interpolation issues to me!

**Comment by mu40** [comment, marked as answer]
Hi @andy9t7! This is an interpolation issue. While you'll want to interpolate the images linearly, doing so with label maps will lead to undesired effects at the label boundaries. You can tell `mri_convert` to use nearest-neighbor interpolation instead:

```sh
mri_convert -rt nearest -ns 1 -at transform.mgz atlas_labelmap.mgz transformed_atlas_labelmap.mgz
```

The argument `-ns  1` ("no scale") avoids renormalizing your voxel intensities, which `mri_convert` does for integer data to use the full range of values.

**Reply by andy9t7** [reply]
Thank you, @mu40 !
